instance/models/instance.py

```diff
@@ -0,0 +1,469 @@
+        """
+        with clone_configuration_repo() as configuration_repo_path:
+            playbook_path = os.path.join(configuration_repo_path, 'playbooks')
+            requirements_path = os.path.join(configuration_repo_path, 'requirements.txt')
+
+            self.log('info', 'Running playbook "{path}/{name}" for instance {instance}...'.format(
+                path=playbook_path,
+                name=playbook_name,
+                instance=self,
+            ))
+
+            log_lines = []
+            with ansible.run_playbook(
+                requirements_path,
+                self.inventory_str,
+                self.vars_str,
+                playbook_path,
+                self.ansible_playbook_filename,
+                username=settings.OPENSTACK_SANDBOX_SSH_USERNAME,
+            ) as processus:
+                for line in processus.stdout:
+                    line = line.decode('utf-8').rstrip()
+                    self.log('info', line)
+                    log_lines.append([line.rstrip()])
+
+        self.log('info', 'Playbook run completed for instance {}'.format(self))
+        return log_lines
+
+
+# Open edX ####################################################################
+
+class OpenEdXInstanceQuerySet(models.QuerySet):
+    """
+    Additional methods for instance querysets
+    Also used as the standard manager for the OpenEdXInstance model
+    """
+    def create(self, *args, **kwargs):
+        """
+        Augmented `create()` method:
+        - Adds support for `fork_name` to allow to set both the github org & repo
+        - Sets the github org & repo to `settings.DEFAULT_FORK` if any is missing
+        - Sets the `commit_id` to the branch tip if it isn't explicitly passed as an argument
+        """
+        fork_name = kwargs.pop('fork_name', None)
+        instance = self.model(**kwargs)
+        if fork_name is None and (not instance.github_organization_name or not instance.github_repository_name):
+            fork_name = settings.DEFAULT_FORK
+        if fork_name is not None:
+            instance.set_fork_name(fork_name, commit=False)
+        if not instance.commit_id:
+            instance.set_to_branch_tip(commit=False)
+        if not instance.name:
+            instance.name = '{i.sub_domain} - {i.reference_name}'.format(i=instance)
+
+        self._for_write = True
+        instance.save(force_insert=True, using=self.db)
+        return instance
+
+    def get(self, *args, **kwargs):
+        """
+        Augmented `get()` method:
+        - Adds support for `fork_name` to allow to query the github org & repo using a single argument
+        """
+        fork_name = kwargs.pop('fork_name', None)
+        if fork_name is not None:
+            kwargs['github_organization_name'], kwargs['github_repository_name'] = fork_name2tuple(fork_name)
+
+        return super().get(*args, **kwargs)
+
+
+class OpenEdXInstance(AnsibleInstanceMixin, GitHubInstanceMixin, LoggerInstanceMixin, Instance):
```

Comment:
@antoviaque The model name shows up as `Opend ed x instance` in the django admin. Not at all important, but I guess there must be some way to give it a nice django admin name; this looks like the setting: https://docs.djangoproject.com/en/1.8/ref/models/options/#verbose-name